src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -172,6 +192,14 @@ public void Pop(bool success)
             else
             {
                 Debug.Assert(_continuationCount == 0);
+
+                if (Current.AsyncEnumerator is not null)
+                {
+                    // we have completed serialization of an AsyncEnumerator,
+                    // pop from the stack and schedule for async disposal.
```

Comment:
why not dispose it now? If we could, would we need to keep track of this list?

Reply:
`IAsyncDisposable` can only be done asynchronously, however `Pop()` is called at the converter context which is sync. We need to store it so that it can be disposed properly in the root async context.